packages/@aws-cdk/aws-config/lib/rule.ts

```diff
@@ -0,0 +1,374 @@
+      public readonly configRuleName = configRuleName;
+
+      public export(): RuleAttributes {
+        return { configRuleName };
+      }
+    }
+
+    return new Import(scope, id);
+  }
+
+  public abstract readonly configRuleName: string;
+
+  /**
+   * Defines a CloudWatch event rule which triggers for rule events. Use
+   * `rule.addEventPattern(pattern)` to specify a filter.
+   */
+  public onEvent(id: string, target?: events.IEventRuleTarget, options?: events.EventRuleProps) {
+    const rule = new events.EventRule(this, id, options);
+    rule.addEventPattern({
+      source: ['aws.config'],
+      detail: {
+        configRuleName: [this.configRuleName]
+      }
+    });
+    rule.addTarget(target);
+    return rule;
+  }
+
+  /**
+   * Defines a CloudWatch event rule which triggers for rule compliance events.
+   */
+  public onComplianceChange(id: string, target?: events.IEventRuleTarget, options?: events.EventRuleProps): events.EventRule {
+    const rule = this.onEvent(id, target, options);
+    rule.addEventPattern({
+      detailType: [ 'Config Rules Compliance Change' ],
+    });
+    return rule;
+  }
+
+  /**
+   * Defines a CloudWatch event rule which triggers for rule re-evaluation status events.
+   */
+  public onReEvaluationStatus(id: string, target?: events.IEventRuleTarget, options?: events.EventRuleProps): events.EventRule {
+    const rule = this.onEvent(id, target, options);
+    rule.addEventPattern({
+      detailType: [ 'Config Rules Re-evaluation Status' ],
+    });
+    return rule;
+  }
+}
+
+/**
+ * A new managed or custom rule.
+ */
+abstract class RuleNew extends RuleBase {
+  public abstract readonly configRuleArn: string;
+  public abstract readonly configRuleId: string;
+  public abstract readonly configRuleComplianceType: string;
+
+  protected scope?: CfnConfigRule.ScopeProperty;
+  protected isManaged?: boolean;
+  protected isCustomWithChanges?: boolean;
+
+  /**
+   * Restrict scope of changes to a specific resource.
+   *
+   * @see https://docs.aws.amazon.com/config/latest/developerguide/resource-config-reference.html#supported-resources
+   *
+   * @param type the resource type
+   * @param identifier the resource identifier
+   */
+  public addResourceScope(type: string, identifier?: string) {
+    this.addScope({
+      complianceResourceId: identifier,
+      complianceResourceTypes: [type],
+    });
+  }
+
+  /**
+   * Restrict scope of changes to specific resource types.
+   *
+   * @see https://docs.aws.amazon.com/config/latest/developerguide/resource-config-reference.html#supported-resources
+   *
+   * @param types resource types
+   */
+  public addResourcesScope(...types: string[]) {
+    this.addScope({
+      complianceResourceTypes: types
+    });
+  }
+
+  /**
+   * Restrict scope of changes to a specific tag.
+   *
+   * @param key the tag key
+   * @param value the tag value
+   */
+  public addTagScope(key: string, value?: string) {
+    this.addScope({
```

Comment:
Maybe a check to confirm that resource and tag scopes aren't mixed?

Reply:
Can you clarify?

Currently the last call to `addXxx()` sets the scope. This is to allow setting the scope even if it's already set by a managed rule, e.g. in `CloudFormationStackNotificationCheck` the scope is set to `AWS::CloudFormation::Stack` but a user could want to restrict it to a specific resource or tag by calling `.addXxx()`:

```ts
const stackNotificationsCheck = new CloudFormationStackNotificationCheck(this, 'Check'); // The scope is set to AWS::CloudFormation::Stack
stackNotificationsCheck.addTagScope('stage', 'prod'); // Changed the scope stage=prod
// addScope() of RuleNew has now been called twice
```

Reply:
I see. The `addXxx` name confused me, seemed like they would be cumulative. I assume this is because jsii won't let you name a method like `setXxx`, right?

I think I would rather have names like `scopeToTag()`, `scopeToResource()` etc. then.

Reply:
Yes exactly, tried with `setXxx`. OK for `scopeToXxx`.